### DISCUSSION TITLE
How To Change Select Options / Default Value

### DISCUSSION THREAD

**Discussion by fweth** [react-hook-form/react-hook-form#8284] [Q&A] [answered]
Hello, I want to include a select item in my form with data that is loaded async.  So in the beginning my `defaultValue` is `""` but after the data has loaded it changes to something else.  Yet when I hit submit, it will submit the original default value.  How to change this?

**Comment by Moshyfawn** [comment, marked as answer]
Create a side-effect using `useEffect` and `reset` the form with your async values when they come in.

**Reply by fweth** [reply]
Thanks a lot!!

**Reply by fweth** [reply]
I just played around a bit and in my case I also like the solution to do `...(data && register(inputName, options))` so that the relevant selections are only registered after they received their data.

**Reply by fweth** [reply]
@Moshyfawn I have another scenario, which is similar, but not directly connected to async loading.  I have two dropdowns A and B, and want the content of B to depend on the current value of A.  When I use `watch` to check for the value of `A` and render the options of `B` accordingly, RHF doesn't update its internal state automatically.  That makes sense, but it would be great to have an option for `resetField` where instead of supplying a `defaultValue`, it just takes the current value from its internal dom reference, that should be possible, I think?  Then I would have a single source of truth and could just write `resetField` with that option in a `useEffect`.

Even nicer would be an function returned from `useForm` which says (to a specific input) "the UI will update with the next render, please update your internal state accordingly" but I don't know if that's technically possible with React.
